integration/js/pages/AppPage.js

```diff
@@ -328,6 +329,21 @@ class AppPage {
     return 'failed'
   }
 
+  async waitForPageToLoad() {
+    let timeout = 10;
+    let i = 0;
+    let loading = false;
+    while (!loading && i < timeout) {
```

Comment:
We reuse this pattern a lot in this code base. Can we abstract it out to a helper wait function? :D

Reply:
Yeah I'll go ahead and add that